File: WordPress/Classes/ViewRelated/System/RotationAwareNavigationViewController.m
#import "RotationAwareNavigationViewController.h"
#import "WordPress-Swift.h"

@implementation RotationAwareNavigationViewController

- (BOOL)shouldAutorotate
{
    // Respect the top child's orientation prefs.
    if (self.topViewController && [self.topViewController respondsToSelector:@selector(shouldAutorotate)]) {
        return [self.topViewController shouldAutorotate];
    }
    
    return [super shouldAutorotate];
}

- (UIInterfaceOrientationMask)supportedInterfaceOrientations
{
    // Respect the top child's orientation prefs.
    if (self.topViewController && [self.topViewController respondsToSelector:@selector(supportedInterfaceOrientations)]) {
        return [self.topViewController supportedInterfaceOrientations];
    }
    
    if ([UIDevice isPad]) {
        return UIInterfaceOrientationMaskAll;
    }
    return UIInterfaceOrientationMaskAllButUpsideDown;
}


@end

File: WordPress/Classes/ViewRelated/System/RotationAwareNavigationViewController.swift
import UIKit
import WordPressShared


class RotationAwareNavigationViewController: UINavigationController {

    override var shouldAutorotate: Bool {
        // Respect the top child's orientation prefs.
        guard let shouldAutorotate = topViewController?.shouldAutorotate else {
            return super.shouldAutorotate
        }

        return shouldAutorotate
    }

    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
        // Respect the top child's orientation prefs.
        if let supportedOrientations = topViewController?.supportedInterfaceOrientations {
            return supportedOrientations
        }

        return WPDeviceIdentification.isiPad() ? .all : .allButUpsideDown
    }
}

